GitUI/UserControls/NativeTreeViewDoubleClickDecorator.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows.Forms;
+
+namespace GitUI.UserControls
+{
+    // Add double-click related behaviour to NativeTreeView
+    public class NativeTreeViewDoubleClickDecorator
+    {
+        private NativeTreeView _treeView;
```

Comment:
```suggestion
        private readonly NativeTreeView _treeView;
```

Reply:
I missed that. However I find this proposed behaviour surprising - an instance of a tree has been created somewhere else and passed to me, why I am responsible for disposing it?
What if I dispose it and there is another consumer still using it?

Reply:
I think this is just me being a C++ programmer, wanting to make sure events get unregistered. In this case, I believe I don't need OnDisposed at all, and can just remove it all.

Reply:
I think that is a better approach: _if you opened - close it, if you subscribed - unsubscribe, when necessary_